pytorch_lightning/accelerators/accelerator.py

```diff
@@ -159,31 +174,31 @@ def training_step(self, step_kwargs: Dict[str, Union[Any, int]]) -> STEP_OUTPUT:
 
         See :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step` for more details
         """
-        with self.precision_plugin.train_step_context():
+        with self.training_type_plugin.precision_plugin.train_step_context():
```

Comment:
Should the precision plugin own the `step_context` instead? I don't have a strong opinion there.

Reply:
Yep, all steps will be either move to training_type_plugin or precision in 2/n or 3/n